### ISSUE TITLE
rethink csv output format for non-string values

### ISSUE THREAD

**Issue by matzemathics** [knowsys/nemo#291]
currently the program

```
p(1, 3).
```

results in 
```
"""1""^^<http://www.w3.org/2001/XMLSchema#integer>","""3""^^<http://www.w3.org/2001/XMLSchema#integer>"
```

which might be technically correct, but seems hard to work with, and we cannot even parse this as a relation of `[integer, integer]` ourselves as in

```
@source p[integer, integer]: load-csv("results/p.csv").
```

which results in no tuples being obtained for p.

Questions:
- is the output what we want here
- (when) should we drop type information when writing to csv
- which explicit syntax of specifying what appears in the output should be supported
- why can't we currently parse the output

**Comment by mmarx** [comment]
> why can't we currently parse the output

we could, if the type were `any` rather than `integer`. Not parsing RDF literal syntax in non-`any`-columns is an explicit choice that we made.

> which explicit syntax of specifying what appears in the output should be supported

We might want to support something like `@output p[integer, any] .` (in fact, the parser might quite likely already support that), and interpret it such that the output becomes `1,"""3""^^<http://www.w3.org/2001/XMLSchema#integer>"`.

**Comment by matzemathics** [comment]
> We might want to support something like @output p[integer, any] . (in fact, the parser might quite likely already support that), and interpret it such that the output becomes 1,"""3""^^<http://www.w3.org/2001/XMLSchema#integer>".

Yes agreed, though this means this issue is blocked by #245

**Comment by mmarx** [comment]
I guess we would also like to write not only CSV, but, e.g., N-Triples or Turtle, so we will need some syntax for specifying even more information for predicates in `@output` directives.

On the other hand, we likely _do_ want a version of CSV output that uses plain numeric literals even for `any` columns (and maybe this should also be the default).

**Comment by monsterkrampe** [comment]
If you add `@declare p(integer, integer).` to the first program, then the output should be written as plain integers.

**Comment by monsterkrampe** [comment]
We will postpone this until #293 is finished.

**Comment by mmarx** [comment]
Let's tackle this once the `DataValue` refactor has landed.

**Comment by mkroetzsch** [comment]
I think this issue is no longer valid as stated since Nemo v0.5.0. The default way of formatting integer values in CSV exports would certainly result in something like `42` rather than the long format that was mentioned above.

Of course, one can still "rethink" how certain values should be formatted in CSV, but this should be raised in individual issues with specific descriptions.
